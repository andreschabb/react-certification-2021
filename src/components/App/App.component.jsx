import React from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';

import AuthProvider from '../../providers/Auth';
import FavoritesPage from '../../pages/Favorites';
import HomePage from '../../pages/Home';
import NotFound from '../../pages/NotFound';
import ResultsPage from '../../pages/Results';
import Layout from '../Layout';
import Private from '../Private';
import GlobalStyles from '../../global.style';
import WatchPage from '../../pages/Watch/Watch.page';

const App = () => {
  return (
    <BrowserRouter>
      <AuthProvider>
        <GlobalStyles />
        <Layout>
          <Switch>
            <Route exact path="/">
              <HomePage />
            </Route>
            <Private exact path="/favorites">
              <FavoritesPage />
            </Private>
            <Route exact path="/results">
              <ResultsPage />
            </Route>
            <Route exact path="/watch">
              <WatchPage />
            </Route>
            <Route path="*">
              <NotFound />
            </Route>
          </Switch>
        </Layout>
      </AuthProvider>
    </BrowserRouter>
  );
};

export default App;
