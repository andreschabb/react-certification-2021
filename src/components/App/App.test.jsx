import React from 'react';
import { render, screen } from '@testing-library/react';

import App from './App.component';

describe('App', () => {
  it('renders App component', () => {
    const { container } = render(<App />);

    expect(screen.getByTestId('layout-sidebar')).toBeTruthy();
    expect(screen.getByTestId('header-menu')).toBeTruthy();

    expect(container).toMatchSnapshot();
  });
});
