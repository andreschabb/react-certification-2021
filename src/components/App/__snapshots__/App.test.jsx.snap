// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App component 1`] = `
<div>
  <div
    class="pushable"
    data-testid="layout-sidebar"
  >
    <div
      class="ui vertical labeled icon ui overlay left thin sidebar menu"
    >
      <div
        class="sc-hHftDr sc-dmlrTW eFfKuu hmZLfg"
      >
        <a
          class="item"
          data-testid="layout-login"
        >
          Login
        </a>
      </div>
      <div
        class="sc-hHftDr sc-kfzAmx eFfKuu fAqkOV"
      >
        <a
          class="item"
          data-testid="layout-dark-mode"
        >
          Dark Mode
        </a>
      </div>
      <a
        class="item"
        data-testid="layout-go-home"
      >
        Home
      </a>
    </div>
    <div
      class="pusher"
      style="height: 100%;"
    >
      <div
        class="ui menu"
        data-testid="header-menu"
      >
        <div
          class="item"
        >
          <i
            aria-hidden="true"
            class="bars icon clickable"
            data-testid="header-bars-icon"
          />
        </div>
        <div
          class="item searchInput"
        >
          <div
            class="ui icon input"
            data-testid="header-search"
          >
            <input
              placeholder="Search..."
              type="text"
              value=""
            />
            <i
              aria-hidden="true"
              class="search icon"
            />
          </div>
        </div>
        <div
          class="right menu"
        >
          <div
            class="item darkMode"
          >
            <div
              class="ui slider checkbox"
              data-testid="header-dark-mode"
            >
              <input
                class="hidden"
                readonly=""
                tabindex="0"
                type="checkbox"
                value=""
              />
              <label>
                Dark Mode
              </label>
            </div>
          </div>
          <div
            aria-expanded="false"
            class="ui dropdown link icon item userIcon"
            data-testid="header-dropdown"
            role="listbox"
            tabindex="0"
          >
            <i
              aria-hidden="true"
              class="user circle outline icon"
            />
            <div
              class="menu transition"
            >
              <div
                class="item"
                data-testid="header-dropdown-login"
                role="option"
              >
                Login
              </div>
            </div>
          </div>
        </div>
      </div>
      <main
        class="sc-dIUggk hpnRtl"
      >
        <section
          class="sc-pFZIQ hewDxE"
        >
          <h1
            class="sc-jrAGrp cgMJfD"
            data-testid="home-message"
          >
            Welcome to the Challenge!
          </h1>
        </section>
      </main>
    </div>
  </div>
</div>
`;
